### PR TITLE
Plot: remove legend argument in backends

### PR DESCRIPTION
**Pull request by t20100**
Merge PR #2730 and #2731 first!

This PR removes the now useless `legend` argument in the plot backend.
This does NOT change the `PlotWidget` and its items, it just changes the backends.

### CODE REVIEW

File: silx/gui/plot/backends/BackendBase.py [hunk 1]
```diff
@@ -263,7 +258,7 @@ def addMarker(self, x, y, legend, text, color,
                           and that returns the filtered coordinates.
         :return: Handle used by the backend to univocally access the marker
         """
-        return legend
+        return object()
```
Comment (vallsv) [inline silx/gui/plot/backends/BackendBase.py]: What about `raise NotImplementedError()`, `BaseBackend` is an abstract class no? It have to be implemented.

**Comment by vallsv** [comment]
LGTM. Looks to be a good idea.

File: silx/gui/plot/backends/BackendBase.py (hunk 1 above)
Comment (t20100) [reply silx/gui/plot/backends/BackendBase.py]: The `BackendBase` can be use to run the `PlotWidget` headless. This is done in the tests at some point.
Basically all methods of `BackendBase` should raise `NotImplementedError`.
Doing it this way provides a mock and the definition of the interface at once.

**Comment by t20100** [comment]
Moving features from the backends to `PlotWidget` allows to simplify the backends.

File: silx/gui/plot/backends/BackendBase.py (hunk 1 above)
Comment (vallsv) [reply silx/gui/plot/backends/BackendBase.py]: Ok.

**Pull request merged**